xoto3/dynamodb/write_versioned/modify.py

```diff
@@ -71,14 +71,17 @@ def _write(
     )
 
     hashable_item_key = hashable_key(item_key)
+    if hashable_item_key in items and items[hashable_item_key] == item_or_none:
+        """You've asked us to write an effect that would have no effect, and we are dropping it"""
```

Comment:
an ideal language would allow these sorts of docstrings to automatically be logs. In fact, all logs would be docstrings - there would be no "library" for logging - the system would simply be aware of an active logger and pass these bare strings into the logger for processing.